Smil

dan-mckay opened the pull request:
take a look at this Dario, it will need some tweaks, no doubt. The tests are running through tape-run2 in Chrome because of the DOMParser issue in Phantom.

Commit:
feat(smil): rewrite smil parser functions to use promises

the tests for smil-data still need to be written after refactoring

Commit:
wip(smil) refactoring smil-data

dan-mckay committed:
wip(smil) optimised parser by filtering out nodes containing whitespace

Commit:
feat(smil): added tests for smil-data and refactored

dariocravero commented on extract/smil-data.es6:
```diff
@@ -11,29 +12,28 @@ const VERSION = 'version';
 
 function extractChildren(xml) {
-  return Array.prototype.map.call(xml.childNodes, extractAttributes).filter(c => !!c);
+  const filteredNodes = Array.prototype.filter.call(xml.childNodes, canIgnoreNode);
```
Once you do a filter you already have an array, so you can just use map over it. Then the code above becomes:

```
return Array.prototype.filter.call(xml.childNodes, canIgnoreNode).map(extractAttributes);
```

Commit:
style(smil): use implicit return

dariocravero commented on extract/smil.es6:
```diff
@@ -36,5 +36,7 @@ export function parseAll(items, manifest, metadata) {
-  return fetchAll(uri, items, manifest).then(parseAll(items, manifest, metadata));
+  return fetchAll(uri, items, manifest)
+    .then(values => parseAll(items, manifest, metadata)(values))
+    .catch(error => console.error(error));
```
We may want to do something with those errors later on. Let's wait until we use this and decide, shall we?

Commit:
chore(smil) adding ignore node file to repo

Commit:
feat: add smil to global tests and add playground

Commit:
fix(smil): arguments in extractSmilData

Commit:
refactor(smil): flatten structure to body on root

Commit:
refactor(smil): simplify data structure on smil

A smil data structure now looks like:
```
{
  ...,
  body: {
    ...
    childNodes: [{
      ...,
      childNodes: [{
        audio: {
          ...
        },
        isPar: true,
        text: {
          ...
        }
      }],
      isPar: false
    }],
    isPar: false
  }
}
```
